x-pack/plugins/cases/server/connectors/cases/cases_connector.ts

```diff
@@ -418,6 +458,115 @@ export class CasesConnector extends SubActionConnector<
       .join(' and ');
   }
 
+  private async handleClosedCases(
+    params: CasesConnectorRunParams,
+    casesClient: CasesClient,
+    casesMap: Map<string, GroupedAlertsWithCases>
+  ) {
+    const entriesWithClosedCases = Array.from(casesMap.values()).filter(
+      (theCase) => theCase.theCase.status === CaseStatuses.closed
+    );
+
+    if (entriesWithClosedCases.length === 0) {
+      return casesMap;
+    }
+
+    const res = params.reopenClosedCases
+      ? await this.reopenClosedCases(casesClient, entriesWithClosedCases, casesMap)
+      : await this.createNewCasesOutOfClosedCases(
+          params,
+          casesClient,
+          entriesWithClosedCases,
+          casesMap
+        );
+
+    /**
+     * The initial map contained the closed cases. We need to remove them to
+     * avoid attaching alerts to a close case
+     */
+    return new Map([...res].filter(([_, record]) => record.theCase.status !== CaseStatuses.closed));
```

Comment:
Yes because the `casesMap` does contain all cases including the closed ones. The `reopenClosedCases` will reopen them but the `createNewCasesOutOfClosedCases` will create new cases without removing the closed ones.